util/self_test.py

```diff
@@ -0,0 +1,38 @@
+import os
+import threading
+from time import sleep
+
+from util.event_stats import get_event_stats_snapshot
+from util.detail import LOGGER
+from main_window.main import LABLES_UPDATED_EVENT, MAP_UPDATED_EVENT
+
+class SelfTest:
+    """
+    The self test is not intended as a thorough test of all functionality. It just provides an automated method to
+    ensure that the GS is able to start without crashing and run basic functionality. Specifically so that CI can
+    automatically test that the PyInstaller builds work.
+    """
+
+    def __init__(self):
+        self.thread = threading.Thread(target=self._run_self_test, daemon=True, name="SelfTestThread")
+
+    def start(self):
+        self.thread.start()
+
+    def _run_self_test(self):
+        try:
+            LOGGER.warning("SELF TEST STARTED")
```

Comment:
I would've thought that warnings would be reserved for things that are probably bad...

Reply:
It's bad if you see this log when you're not expecting it. You're right, I know what you're saying, but if we ever filter our logs/console by log level, we might want this to still shows up.

I'd be happy to demote it to an info log if you think that's more appropriate.